test/integration/associations/belongs-to-many.test.js

```diff
@@ -357,30 +364,61 @@ describe(Support.getTestDialectTeaser('BelongsToMany'), () => {
       }), Group.findAll({
         include: [User]
       })]);
-
+      //Need to add db2 condition for the same. referred to issue: https://github.com/chaijs/chai/issues/102
       expect(users.length).to.be.equal(2);
       expect(users[0].Groups.length).to.be.equal(1);
       expect(users[1].Groups.length).to.be.equal(1);
       expect(users[0].Groups[0].usergroups.UserUserSecondId).to.be.ok;
-      expect(users[0].Groups[0].usergroups.UserUserSecondId).to.be.equal(users[0].userSecondId);
+      if (dialect === 'db2') {
+        expect(users[0].Groups[0].usergroups.UserUserSecondId).to.deep.equal(users[0].userSecondId);
```

Comment:
I think we could just go with deep equal by default. However, what are we getting here and is that what you expect?